### ISSUE TITLE
Golang update PRs sometimes don't update go.sum

### ISSUE THREAD

**Issue by djmitche** [renovatebot/config-help#651]
### Which Renovate are you using?

WhiteSource Renovate App

### Which platform are you using?

GitHub.com

### Have you checked the logs? Don't forget to include them if relevant

I have - they are [here](https://app.renovatebot.com/dashboard#github/taskcluster/taskcluster/175595315).  I'm not sure what to make of them.  It looks like renovate updated `go.mod`
```
DEBUG: gomod: need to update digest(branch="renovate/golang.org-x-net-digest")
{
  "depName": "golang.org/x/net",
  "lineToChange": "\tgolang.org/x/net v0.0.0-20200505041828-1ed23360d12c",
  "newDigestRightSized": "7e3656a0809f"
}
```
then ran `go get` and `go mod tidy`.

### What would you like to do?

Our own CI runs on PRs and checks that `go mod tidy` has been run by looking for a diff.  In [this case](https://github.com/taskcluster/taskcluster/pull/2795) it found:
```diff
diff --git a/go.sum b/go.sum
index 8c53a8ce1..2eec04d6e 100644
--- a/go.sum
+++ b/go.sum
@@ -227,6 +227,7 @@ golang.org/x/net v0.0.0-20190522155817-f3200d17e092/go.mod h1:HSz+uSET+XFnRR8LxR
 golang.org/x/net v0.0.0-20190620200207-3b0461eec859/go.mod h1:z5CRVTTTmAJ677TzLLGU+0bjPO0LkuOLi4/5GtJWs/s=
 golang.org/x/net v0.0.0-20200202094626-16171245cfb2/go.mod h1:z5CRVTTTmAJ677TzLLGU+0bjPO0LkuOLi4/5GtJWs/s=
 golang.org/x/net v0.0.0-20200226121028-0de0cce0169b/go.mod h1:z5CRVTTTmAJ677TzLLGU+0bjPO0LkuOLi4/5GtJWs/s=
+golang.org/x/net v0.0.0-20200506145744-7e3656a0809f h1:QBjCr1Fz5kw158VqdE9JfI9cJnl/ymnJWAdMuinqL7Y=
 golang.org/x/net v0.0.0-20200506145744-7e3656a0809f/go.mod h1:qpuaurCH72eLCgpAm/N6yyVIVM9cpaDIP3A8BGJEC5A=
 golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be/go.mod h1:N/0e6XlmueqKjAGxoOufVs8QHGRruUQn6yWY3a++T0U=
 golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f/go.mod h1:RxMgew5VJxzue5/jJTE5uejpjVlOe/izrB70Jof72aM=
```

This has happened quite a bit in various PRs.  In most cases, `rebase!` will fix it, but of course that means waiting an hour and running more CI tasks.  We do have
```
    "postUpdateOptions": [
      "gomodTidy"
    ]   
```
in the config.

**Comment by rarkins** [comment]
Is this the PR that you're referring to?

![image](https://user-images.githubusercontent.com/6311784/81708895-d42aa000-9471-11ea-8b06-601196eea8df.png)

If so, this issue title is a bit confusing, or am I missing something?

What it looks like to me is:

- Renovate did run `go mod tidy`
- It *did* do some tidying (note the removed lines)
- Your CI reports that *more* tidying is needed, but we don't know why unless `go` has a bug that requires you to tidy more than once

Is this accurate?

**Comment by rarkins** [comment]
> but of course that means waiting an hour and running more CI tasks.

Rebases should happen almost immediately once you request them. I went back through the repo's closed PRs and the first one I found where you'd renamed it showed Renovate force pushed 5 minutes later. Do you have any recent examples of it taking too long?

**Comment by djmitche** [comment]
I think what's missing from the renovate run is the second line of `go.sum` that refers to the package name without the `/go.mod` suffix.  Whether that line was added by `go get` and removed by `go mod tidy` or just never added is unclear.

It does seem likely it's a Go bug (go mod support is still pretty janky in general).  Maybe running twice would help?

It's waiting for the CI to finish that's about an hour -- not renovate's fault :)

**Comment by rarkins** [comment]
Yeah.. I think maybe a double `go mod tidy` is in order. We already run it twice in a single PR in one other scenario (go mod tidy / go mod vendor / go mod tidy IIRC)
